benchmark_utils/optuna_solver.py

```diff
@@ -19,17 +21,27 @@ class OSolver(BaseSolver):
 
     stopping_criterion = SufficientProgressCriterion(strategy='callback')
 
+    params = {
+        'test_size': 0.25,
+        'seed': 42,
+    }
+
     def set_objective(
-            self, X_train, y_train, X_test, y_test,
+            self, X_train, y_train,
             categorical_indicator
     ):
         # Define the information received by each solver from the objective.
         # The arguments of this function are the results of the
         # `Objective.get_objective`. This defines the benchmark's API for
         # passing the objective to the solver.
         # It is customizable for each benchmark.
-        self.X_train, self.y_train = X_train, y_train
-        self.X_test, self.y_test = X_test, y_test
+        rng = np.random.RandomState(self.params['seed'])
+        X, X_val, y, y_val = train_test_split(
+            X_train, y_train, test_size=self.params['test_size'],
+            random_state=rng
+        )

```

Comment:
A simpler way actually is:
```suggestion
        X, X_val, y, y_val = train_test_split(
            X_train, y_train, test_size=self.params['test_size'],
            random_state=self.params['seed']
        )

```